### PR TITLE
Improve safe dump

### PR DESCRIPTION
**Pull request by hermansje**
Warn when not returning expected format

cc @dfournier @nickvdyck

### CODE REVIEW

File: protobackend/output.py [hunk 1]
```diff
@@ -55,10 +59,15 @@ def wrapper(*args, **kwargs):
         with CaptureErrors(fallback_output):
             output = f(*args, **kwargs)
             if not isinstance(output, list):
+                if output is not None:
```
Comment (dfournier) [inline protobackend/output.py]: If I get it right, `None` returns an empty array but that's a wanted behavior? I suggest either to log in that case or mention is on the description.

File: protobackend/output.py [hunk 2]
```diff
@@ -55,10 +59,15 @@ def wrapper(*args, **kwargs):
...
+                    warnings.warn(
+                        "Only lists are valid return values for executed commands"
```
Comment (dfournier) [inline protobackend/output.py]: Here I guess `only lists and None` is more appropriate?

**Review by davidfou — COMMENTED** [review]
Do you think it worth it to add tests on your first commit about `safe_dump`?

File: protobackend/output.py (hunk 1 above)
Comment (hermansje) [reply protobackend/output.py]: `None` is the implicit return value, which allows commands that just do something without output to not have `return []`.

File: protobackend/output.py (hunk 2 above)
Comment (hermansje) [reply protobackend/output.py]: maybe `only lists and (implicit) None`?

Comment (dfournier) [reply protobackend/output.py]: This comment is actually linked to the other one. So in Python `return None` or no return gives the same output? If so, maybe `"Executed commands are expected to return a list or nothing"`. Do we want to mention the output has been stripped out to an empty array?

Comment (hermansje) [reply protobackend/output.py]: Yes, that's what Python does! I updated the message.

**Comment by dfournier** [comment]
Just to be sure, no test has been added about `safe_dump` function?

**Comment by dfournier** [comment]
@hermansje did you see my comment? I think you forgot to add unit tests

**Review by davidfou — COMMENTED** [review]
No unit test? 😢

**Comment by hermansje** [comment]
No, no time 😢

**Comment by dfournier** [comment]
No approval then!

**Comment by TimSangster** [comment]
🎉

**Review by davidfou — APPROVED** [review]

**Pull request merged**